File: puntoDeVenta/pos-core/src/main/java/com/empresa/pos/service/CartService.java
package com.empresa.pos.service;

import java.math.BigDecimal;
import java.util.Collection;

public class CartService {

    private static final BigDecimal IVA_RATE = new BigDecimal("0.16");

    public BigDecimal calcularSubtotal(Collection<BigDecimal> subtotales) {
        return subtotales.stream().reduce(BigDecimal.ZERO, BigDecimal::add);
    }

    public BigDecimal calcularIva(BigDecimal subtotal) {
        return subtotal.multiply(IVA_RATE);
    }

    public BigDecimal calcularTotal(BigDecimal subtotal) {
        return subtotal.add(calcularIva(subtotal));
    }
}

File: pos-maui/Modelo/Servicio/CartService.cs
namespace PosMaui.Modelo.Servicio;

public static class CartService
{
    private const decimal IVA = 0.16m;

    public static decimal CalcularSubtotal(IEnumerable<decimal> subtotales)
        => subtotales.Sum();

    public static decimal CalcularIva(decimal subtotal)
        => subtotal * IVA;

    public static decimal CalcularTotal(decimal subtotal)
        => subtotal + CalcularIva(subtotal);
}
